File: src/test/java/com/doestheohavethisbook/AuthorServiceTests.java
package com.doestheohavethisbook;

import org.easymock.EasyMockRunner;
import org.easymock.EasyMockSupport;
import org.easymock.Mock;
import org.easymock.TestSubject;
import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.List;
import java.util.UUID;

import static org.easymock.EasyMock.expect;
import static org.junit.Assert.assertEquals;

@RunWith(EasyMockRunner.class)
public class AuthorServiceTests extends EasyMockSupport {
    @Mock
    private AuthorRepository repository;

    @TestSubject
    private final AuthorService service = new AuthorServiceImpl();

    @Test
    public void whenGetAllAuthorsCalled_thenCallsRepository() {
        Author author1 = new Author("Bob", "Martin", UUID.randomUUID());
        Author author2 = new Author("Martin", "Fowler", UUID.randomUUID());
        List<Author> expectedAuthors = List.of(author1, author2);
        expect(repository.findAll()).andReturn(expectedAuthors);
        replayAll();

        List<Author> result = service.getAllAuthors();

        assertEquals(expectedAuthors, result);
        verifyAll();
    }
}

File: src/test/java/com/doestheohavethisbook/AuthorServiceTests.kt
package com.doestheohavethisbook

import io.mockk.every
import io.mockk.mockk
import io.mockk.verify
import junit.framework.TestCase.assertEquals
import org.junit.Test

import java.util.UUID

internal class AuthorServiceTests {
    private val repository = mockk<AuthorRepository>()

    @Test
    fun whenGetAllAuthorsCalled_thenCallsRepository() {
        val author1 = Author(UUID.randomUUID(), "Bob", "Martin")
        val author2 = Author(UUID.randomUUID(), "Martin", "Fowler")
        val expectedAuthors = listOf(author1, author2)
         every { repository.findAll() }
             .returns(expectedAuthors)
        val service = AuthorService()
        service.repository = repository

        val result = service.getAllAuthors()

        assertEquals(expectedAuthors, result)
        verify { repository.findAll() }
    }
}
